keps/sig-scheduling/5194-reserved-for-workloads/README.md

```diff
@@ -0,0 +1,796 @@
+is scheduled to use the `ResourceClaim`. It is possible to have pods that reference `ResourceClaim`,
+but haven't yet been scheduled. This distinction is important for some of the usages of the
+`ReservedFor` list described above:
+
+1. If the kubelet sees that the `status.allocation.ReservedForAnyPod` is set, it will skip
+   the check that the Pod is listed in the `ReservedFor` list and just run the pod.
+
+1. If the DRA scheduler plugin is trying to find candidates for deallocation in
+   the `PostFilter` function and sees a `ResourceClaim` with a non-Pod reference, it will not
+   attempt to deallocate. The plugin has no way to know how many pods are actually consuming
+   the `ResourceClaim` without the explit list in the `ReservedFor` list and therefore it will
+   not be safe to deallocate.
+
+1. The device_taint_eviction controller will use the list of pods referencing the `ResourceClaim`
+   to determine the list of pods that needs to be evicted. In this situation, it is ok if the
+   list includes pods that haven't yet been scheduled.
+
+
+### Test Plan
+
+<!--
+**Note:** *Not required until targeted at a release.*
+The goal is to ensure that we don't accept enhancements with inadequate testing.
+
+All code is expected to have adequate tests (eventually with coverage
+expectations). Please adhere to the [Kubernetes testing guidelines][testing-guidelines]
+when drafting this test plan.
+
+[testing-guidelines]: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
+-->
+
+[ ] I/we understand the owners of the involved components may require updates to
+existing tests to make this code solid enough prior to committing the changes necessary
+to implement this enhancement.
+
+##### Prerequisite testing updates
+
+None
+
+##### Unit tests
+
+<!--
+Generated with:
+
+go test -cover ./pkg/scheduler/framework/plugins/dynamicresources ./pkg/controller/resourceclaim ./pkg/controller/devicetainteviction ./pkg/kubelet/cm/dra | sed -e 's/.*\(k8s.io[a-z/-]*\).*coverage: \(.*\) of statements/- `\1`: \2/' | sort
+
+-->
+
+
+- `k8s.io/kubernetes/pkg/controller/devicetainteviction`: `06/05/2025` - 89.9%
+- `k8s.io/kubernetes/pkg/controller/resourceclaim`: `06/05/2025` - 74.2%
+- `k8s.io/kubernetes/pkg/kubelet/cm/dra`: `06/05/2025` - 79.4%
+- `k8s.io/kubernetes/pkg/scheduler/framework/plugins/dynamicresources`: `06/05/2025` - 79.3%
+
+##### Integration tests
+
+Scheduler perf tests will be added to assess the performance impact of this change.
+
+##### e2e tests
+
+Additional e2e tests will be added to verify the behavior added in this KEP.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature implemented behind a feature flag
+- Initial e2e tests completed and enabled
+
+#### Beta
+
+- Gather feedback from developers and surveys
+- Additional tests are in Testgrid and linked in KEP
+- Performance impact of the feature has been measured and found to be acceptable
+
+#### GA
+
+- 3 examples of real-world usage
```

Comment:
I don't think these are the right criteria. I am not sure what three there would be. Deployment, Job, StatefulSet maybe? Also I think the latest template may have some additional items here.

Reply:
Yeah, I've updated this one. With the new beta requirements, there shouldn't be all that much left for GA.